looks solid to me once tests pass:
```
======================================================================
ERROR: testSubmitBadXML (corehq.apps.receiverwrapper.tests.test_submit_errors.SubmissionErrorTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/receiverwrapper/tests/test_submit_errors.py", line 110, in testSubmitBadXML
    limit=1)
ValueError: need more than 0 values to unpack
----------------------------------------------------------------------
Ran 289 tests in 312.402s
```

--- a/corehq/apps/cleanup/management/commands/rebuild_cases_with_mismatched_forms.py
+++ b/corehq/apps/cleanup/management/commands/rebuild_cases_with_mismatched_forms.py
@@ -98,7 +98,7 @@ def archive_exception(exception):
 
 def should_rebuild(domain, case_id):
     try:
-        case = FormProcessorInterface(domain).case_model.get(case_id)
+        case = FormProcessorInterface(domain).get_case(case_id)
         return hasattr(case, 'actions') and case.actions[-1].action_type != CASE_ACTION_REBUILD
     except CaseNotFound:
         return True

--- a/corehq/apps/hqcase/tests/test_case_assigment.py
+++ b/corehq/apps/hqcase/tests/test_case_assigment.py
@@ -136,13 +136,13 @@ class CaseAssignmentTest(TestCase):
     def _check_state(self, new_owner_id, expected_changed, update=None):
         expected_ids = set(c._id for c in expected_changed)
         for case in expected_changed:
-            expected = FormProcessorInterface().case_model.get(case._id)
+            expected = FormProcessorInterface().get_case(case._id)
             self.assertEqual(new_owner_id, expected.owner_id)
             if update:
                 for prop, value in update.items():
                     self.assertEqual(getattr(expected, prop), value)
         for case in (c for c in self.all if c._id not in expected_ids):
-            remaining = FormProcessorInterface().case_model.get(case._id)
+            remaining = FormProcessorInterface().get_case(case._id)
             self.assertEqual(self.original_owner._id, remaining.owner_id)
 
     def _new_case(self, index=None):

--- a/corehq/ex-submodules/casexml/apps/case/mock.py
+++ b/corehq/ex-submodules/casexml/apps/case/mock.py
@@ -333,6 +333,6 @@ class CaseFactory(object):
             form_extras,
         )
 
-        return FormProcessorInterface().case_model.get_cases(
+        return FormProcessorInterface().get_cases(
             [id for structure in case_structures for id in structure.walk_ids()]
         )

--- a/corehq/ex-submodules/casexml/apps/case/tests/test_delete.py
+++ b/corehq/ex-submodules/casexml/apps/case/tests/test_delete.py
@@ -16,14 +16,14 @@ class TestHardDelete(TestCase):
         factory = CaseFactory()
         case = factory.create_case()
         [case] = factory.create_or_update_case(CaseStructure(case_id=case._id, attrs={'update': {'foo': 'bar'}}))
-        self.assertIsNotNone(self.interface.case_model.get(case.case_id))
+        self.assertIsNotNone(self.interface.get_case(case.case_id))
         self.assertEqual(2, len(case.xform_ids))
         for form_id in case.xform_ids:
             self.assertIsNotNone(self.interface.xform_model.get(form_id))
         case.hard_delete()
 
         with self.assertRaises(CaseNotFound):
-            self.interface.case_model.get(case.case_id)
+            self.interface.get_case(case.case_id)
 
         for form_id in case.xform_ids:
             with self.assertRaises(XFormNotFound):
@@ -43,9 +43,9 @@ class TestHardDelete(TestCase):
 
         # deleting the child is ok
         child.hard_delete()
-        self.assertIsNotNone(self.interface.case_model.get(parent.case_id))
+        self.assertIsNotNone(self.interface.get_case(parent.case_id))
         with self.assertRaises(CaseNotFound):
-            self.interface.case_model.get(child.case_id)
+            self.interface.get_case(child.case_id)
 
     def test_delete_sharing_form(self):
         factory = CaseFactory()
@@ -59,5 +59,5 @@ class TestHardDelete(TestCase):
         with self.assertRaises(CommCareCaseError):
             c2.hard_delete()
 
-        self.assertIsNotNone(self.interface.case_model.get(c1.case_id))
-        self.assertIsNotNone(self.interface.case_model.get(c2.case_id))
+        self.assertIsNotNone(self.interface.get_case(c1.case_id))
+        self.assertIsNotNone(self.interface.get_case(c2.case_id))

--- a/corehq/ex-submodules/casexml/apps/case/tests/test_delete.py
+++ b/corehq/ex-submodules/casexml/apps/case/tests/test_delete.py
@@ -19,7 +19,7 @@ class TestHardDelete(TestCase):
         self.assertIsNotNone(self.interface.get_case(case.case_id))
         self.assertEqual(2, len(case.xform_ids))
         for form_id in case.xform_ids:
-            self.assertIsNotNone(self.interface.xform_model.get(form_id))
+            self.assertIsNotNone(self.interface.get_xform(form_id))
         case.hard_delete()
 
         with self.assertRaises(CaseNotFound):
@@ -27,7 +27,7 @@ class TestHardDelete(TestCase):
 
         for form_id in case.xform_ids:
             with self.assertRaises(XFormNotFound):
-                self.interface.xform_model.get(form_id)
+                self.interface.get_xform(form_id)
 
     def test_delete_with_related(self):
         factory = CaseFactory()

--- a/corehq/ex-submodules/casexml/apps/case/tests/test_factory.py
+++ b/corehq/ex-submodules/casexml/apps/case/tests/test_factory.py
@@ -103,7 +103,7 @@ class CaseFactoryTest(TestCase):
     def test_simple_create(self):
         factory = CaseFactory()
         case = factory.create_case()
-        self.assertIsNotNone(FormProcessorInterface().case_model.get(case.case_id))
+        self.assertIsNotNone(FormProcessorInterface().get_case(case.case_id))
 
     def test_create_overrides(self):
         factory = CaseFactory()

--- a/corehq/ex-submodules/casexml/apps/case/tests/test_factory.py
+++ b/corehq/ex-submodules/casexml/apps/case/tests/test_factory.py
@@ -177,7 +177,7 @@ class CaseFactoryTest(TestCase):
         token_id = uuid.uuid4().hex
         factory = CaseFactory(domain=domain)
         [case] = factory.create_or_update_case(CaseStructure(), form_extras={'last_sync_token': token_id})
-        form = FormProcessorInterface(domain).xform_model.get(case.xform_ids[0])
+        form = FormProcessorInterface(domain).get_xform(case.xform_ids[0])
         self.assertEqual(token_id, form.last_sync_token)
 
     def test_form_extras_default(self):
@@ -188,7 +188,7 @@ class CaseFactoryTest(TestCase):
         token_id = uuid.uuid4().hex
         factory = CaseFactory(domain=domain, form_extras={'last_sync_token': token_id})
         case = factory.create_case()
-        form = FormProcessorInterface(domain).xform_model.get(case.xform_ids[0])
+        form = FormProcessorInterface(domain).get_xform(case.xform_ids[0])
         self.assertEqual(token_id, form.last_sync_token)
 
     def test_form_extras_override_defaults(self):
@@ -197,5 +197,5 @@ class CaseFactoryTest(TestCase):
         token_id = uuid.uuid4().hex
         factory = CaseFactory(domain=domain, form_extras={'last_sync_token': token_id})
         [case] = factory.create_or_update_case(CaseStructure(), form_extras={'last_sync_token': 'differenttoken'})
-        form = FormProcessorInterface(domain).xform_model.get(case.xform_ids[0])
+        form = FormProcessorInterface(domain).get_xform(case.xform_ids[0])
         self.assertEqual('differenttoken', form.last_sync_token)

--- a/corehq/ex-submodules/casexml/apps/case/tests/test_from_xform.py
+++ b/corehq/ex-submodules/casexml/apps/case/tests/test_from_xform.py
@@ -33,7 +33,7 @@ class CaseFromXFormTest(TestCase):
         
         case = bootstrap_case_from_xml(self, "update.xml", original_case.case_id)
         # fetch the case from the DB to ensure it is property wrapped
-        case = self.interface.case_model.get(case.case_id)
+        case = self.interface.get_case(case.case_id)
         self.assertEqual(False, case.closed)
         
         self.assertEqual(3, len(case.actions))

--- a/corehq/ex-submodules/casexml/apps/case/tests/test_multi_case_submits.py
+++ b/corehq/ex-submodules/casexml/apps/case/tests/test_multi_case_submits.py
@@ -41,6 +41,6 @@ class MultiCaseTest(TestCase):
 
     def _check_ids(self, form, cases):
         for case in cases:
-            ids = self.interface.case_model.get_case_xform_ids(case.case_id)
+            ids = self.interface.get_case_xform_ids(case.case_id)
             self.assertEqual(1, len(ids))
             self.assertEqual(form._id, ids[0])

--- a/corehq/ex-submodules/casexml/apps/case/tests/test_multimedia.py
+++ b/corehq/ex-submodules/casexml/apps/case/tests/test_multimedia.py
@@ -203,7 +203,7 @@ class CaseMultimediaTest(BaseCaseMultimediaTest):
         self._validateOTARestore(case.case_id, restore_attachments)
 
     def _validateOTARestore(self, case_id, restore_attachments):
-        case_xml = self.interface.case_model.get(case_id).to_xml(V2)
+        case_xml = self.interface.get_case(case_id).to_xml(V2)
         root_node = lxml.etree.fromstring(case_xml)
         attaches = root_node.find('{http://commcarehq.org/case/transaction/v2}attachment')
         self.assertEqual(len(restore_attachments), len(attaches))

--- a/corehq/ex-submodules/casexml/apps/case/tests/test_multimedia.py
+++ b/corehq/ex-submodules/casexml/apps/case/tests/test_multimedia.py
@@ -156,14 +156,14 @@ class CaseMultimediaTest(BaseCaseMultimediaTest):
         _, _, [case] = self._doCreateCaseWithMultimedia(attachments=[single_attach])
 
         for xform_id in case.xform_ids:
-            form = self.interface.xform_model.get(xform_id)
+            form = self.interface.get_xform(xform_id)
 
             form.archive()
-            form = self.interface.xform_model.get(xform_id)
+            form = self.interface.get_xform(xform_id)
             self.assertTrue(form.is_archived)
 
             form.unarchive()
-            form = self.interface.xform_model.get(xform_id)
+            form = self.interface.get_xform(xform_id)
             self.assertFalse(form.is_archived)
 
     def testAttachRemoveSingle(self):

--- a/corehq/ex-submodules/casexml/apps/case/tests/test_out_of_order_processing.py
+++ b/corehq/ex-submodules/casexml/apps/case/tests/test_out_of_order_processing.py
@@ -19,7 +19,7 @@ class OutOfOrderCaseTest(TestCase):
                 xml_data = f.read()
             interface.submit_form_locally(xml_data)
 
-        case = interface.case_model.get('30bc51f6-3247-4966-b4ae-994f572e85fe')
+        case = interface.get_case('30bc51f6-3247-4966-b4ae-994f572e85fe')
         self.assertEqual('from the update form', case.pupdate)
         self.assertEqual('from the create form', case.pcreate)
         self.assertEqual('overridden by the update form', case.pboth)

--- a/corehq/ex-submodules/casexml/apps/phone/tests/test_sync_mode.py
+++ b/corehq/ex-submodules/casexml/apps/phone/tests/test_sync_mode.py
@@ -690,7 +690,7 @@ class SyncDeletedCasesTest(SyncBaseTest):
                 )],
             )
         ])
-        FormProcessorInterface().case_model.get(parent_id).soft_delete()
+        FormProcessorInterface().get_case(parent_id).soft_delete()
         assert_user_doesnt_have_case(self, self.user, parent_id)
         # todo: in the future we may also want to purge the child
         assert_user_has_case(self, self.user, child_id)

--- a/corehq/ex-submodules/couchforms/tests/test_archive.py
+++ b/corehq/ex-submodules/couchforms/tests/test_archive.py
@@ -37,7 +37,7 @@ class TestFormArchiving(TestCase, TestFileMixin):
 
         xform = self.interface.xform_model.get(xform.form_id)
         self.assertTrue(xform.is_archived)
-        case = self.interface.case_model.get(case_id)
+        case = self.interface.get_case(case_id)
         self.assertTrue(case.is_deleted)
         self.assertEqual(case.xform_ids, [])
 
@@ -52,7 +52,7 @@ class TestFormArchiving(TestCase, TestFileMixin):
 
         xform = self.interface.xform_model.get(xform.form_id)
         self.assertTrue(xform.is_normal)
-        case = self.interface.case_model.get(case_id)
+        case = self.interface.get_case(case_id)
         self.assertFalse(case.is_deleted)
         self.assertEqual(case.xform_ids, [xform.form_id])
 

--- a/corehq/ex-submodules/couchforms/tests/test_archive.py
+++ b/corehq/ex-submodules/couchforms/tests/test_archive.py
@@ -35,7 +35,7 @@ class TestFormArchiving(TestCase, TestFileMixin):
         xform.archive(user='mr. librarian')
         upper_bound = datetime.utcnow() + timedelta(seconds=1)
 
-        xform = self.interface.xform_model.get(xform.form_id)
+        xform = self.interface.get_xform(xform.form_id)
         self.assertTrue(xform.is_archived)
         case = self.interface.get_case(case_id)
         self.assertTrue(case.is_deleted)
@@ -50,7 +50,7 @@ class TestFormArchiving(TestCase, TestFileMixin):
         xform.unarchive(user='mr. researcher')
         upper_bound = datetime.utcnow() + timedelta(seconds=1)
 
-        xform = self.interface.xform_model.get(xform.form_id)
+        xform = self.interface.get_xform(xform.form_id)
         self.assertTrue(xform.is_normal)
         case = self.interface.get_case(case_id)
         self.assertFalse(case.is_deleted)

--- a/corehq/ex-submodules/couchforms/tests/test_duplicates.py
+++ b/corehq/ex-submodules/couchforms/tests/test_duplicates.py
@@ -43,7 +43,7 @@ class DuplicateFormTest(TestCase, TestFileMixin):
 
         # Change the doc_type of the form by archiving it
         self.interface.xform_model.archive(xform1)
-        xform1 = self.interface.xform_model.get(xform1.form_id)
+        xform1 = self.interface.get_xform(xform1.form_id)
         self.assertTrue(xform1.is_archived)
 
         # Post an xform with that has different doc_type but same id

--- a/corehq/ex-submodules/couchforms/tests/test_edits.py
+++ b/corehq/ex-submodules/couchforms/tests/test_edits.py
@@ -129,7 +129,7 @@ class EditFormTest(TestCase, TestFileMixin):
         submit_case_blocks(case_block, domain=self.domain, form_id=form_id)
 
         # validate some assumptions
-        case = self.interface.case_model.get(case_id)
+        case = self.interface.get_case(case_id)
         self.assertEqual(case.type, 'person')
         self.assertEqual(case.dynamic_case_properties()['property'], 'original value')
         self.assertEqual([form_id], case.xform_ids)
@@ -151,7 +151,7 @@ class EditFormTest(TestCase, TestFileMixin):
         ).as_string()
         submit_case_blocks(case_block, domain=self.domain, form_id=form_id)
 
-        case = self.interface.case_model.get(case_id)
+        case = self.interface.get_case(case_id)
         self.assertEqual(case.type, 'newtype')
         self.assertEqual(case.dynamic_case_properties()['property'], 'edited value')
         self.assertEqual([form_id], case.xform_ids)
@@ -201,7 +201,7 @@ class EditFormTest(TestCase, TestFileMixin):
         create_form_id = submit_case_blocks(case_block, domain=self.domain)
 
         # validate that worked
-        case = self.interface.case_model.get(case_id)
+        case = self.interface.get_case(case_id)
         self.assertEqual([create_form_id], case.xform_ids)
 
         if not settings.TESTS_SHOULD_USE_SQL_BACKEND:
@@ -222,7 +222,7 @@ class EditFormTest(TestCase, TestFileMixin):
         edit_form_id = submit_case_blocks(case_block, domain=self.domain)
 
         # validate that worked
-        case = self.interface.case_model.get(case_id)
+        case = self.interface.get_case(case_id)
         self.assertEqual(case.dynamic_case_properties()['property'], 'first value')
         self.assertEqual([create_form_id, edit_form_id], case.xform_ids)
 
@@ -240,7 +240,7 @@ class EditFormTest(TestCase, TestFileMixin):
         second_edit_form_id = submit_case_blocks(case_block, domain=self.domain)
 
         # validate that worked
-        case = self.interface.case_model.get(case_id)
+        case = self.interface.get_case(case_id)
         self.assertEqual(case.dynamic_case_properties()['property'], 'final value')
         self.assertEqual([create_form_id, edit_form_id, second_edit_form_id], case.xform_ids)
 
@@ -264,7 +264,7 @@ class EditFormTest(TestCase, TestFileMixin):
 
         # ensure that the middle edit stays in the right place and is applied
         # before the final one
-        case = self.interface.case_model.get(case_id)
+        case = self.interface.get_case(case_id)
         self.assertEqual(case.dynamic_case_properties()['property'], 'final value')
         self.assertEqual(case.dynamic_case_properties()['added_property'], 'added value')
         self.assertEqual([create_form_id, edit_form_id, second_edit_form_id], case.xform_ids)

--- a/corehq/ex-submodules/couchforms/tests/test_edits.py
+++ b/corehq/ex-submodules/couchforms/tests/test_edits.py
@@ -54,7 +54,7 @@ class EditFormTest(TestCase, TestFileMixin):
         self.assertEqual("100", xform.form_data['vitals']['height'])
         self.assertEqual("Edited Baby!", xform.form_data['assessment']['categories'])
 
-        deprecated_xform = self.interface.xform_model.get(xform.deprecated_form_id)
+        deprecated_xform = self.interface.get_xform(xform.deprecated_form_id)
 
         self.assertEqual(self.ID, deprecated_xform.orig_id)
         self.assertNotEqual(self.ID, deprecated_xform.form_id)
@@ -100,7 +100,7 @@ class EditFormTest(TestCase, TestFileMixin):
         # it didn't go through, so make sure there are no edits still
         self.assertIsNone(getattr(xform, 'deprecated_form_id', None))
 
-        xform = self.interface.xform_model.get(self.ID)
+        xform = self.interface.get_xform(self.ID)
         self.assertIsNotNone(xform)
         self.assertEqual(
             UnfinishedSubmissionStub.objects.filter(xform_id=self.ID,
@@ -180,10 +180,10 @@ class EditFormTest(TestCase, TestFileMixin):
         ).as_string()
         submit_case_blocks(case_block, domain=self.domain, form_id=form_id)
 
-        xform = self.interface.xform_model.get(form_id)
+        xform = self.interface.get_xform(form_id)
         self.assertTrue(xform.is_error)
 
-        deprecated_xform = self.interface.xform_model.get(xform.deprecated_form_id)
+        deprecated_xform = self.interface.get_xform(xform.deprecated_form_id)
         self.assertTrue(deprecated_xform.is_deprecated)
 
     @run_with_all_backends

--- a/corehq/ex-submodules/couchforms/util.py
+++ b/corehq/ex-submodules/couchforms/util.py
@@ -128,8 +128,9 @@ def _handle_duplicate(new_doc, instance):
     and *must* include inline attachments
 
     """
+    interface = FormProcessorInterface(new_doc.domain)
     conflict_id = new_doc.form_id
-    existing_doc = FormProcessorInterface().xform_model.get_with_attachments(conflict_id)
+    existing_doc = interface.get_form_with_attachments(conflict_id)
 
     existing_md5 = existing_doc.xml_md5()
     new_md5 = hashlib.md5(instance).hexdigest()

--- a/corehq/form_processor/interfaces/processor.py
+++ b/corehq/form_processor/interfaces/processor.py
@@ -82,6 +82,12 @@ class FormProcessorInterface(object):
     def post_case_blocks(self, case_blocks, form_extras=None, domain=None):
         return post_case_blocks(case_blocks, form_extras=form_extras, domain=domain)
 
+    def get_xform(self, form_id):
+        return self.xform_model.get(form_id)
+
+    def get_form_with_attachments(self, form_id):
+        return self.xform_model.get_with_attachments(form_id)
+
     def get_case(self, case_id):
         return self.case_model.get(case_id)
 

--- a/corehq/form_processor/interfaces/processor.py
+++ b/corehq/form_processor/interfaces/processor.py
@@ -82,6 +82,9 @@ class FormProcessorInterface(object):
     def post_case_blocks(self, case_blocks, form_extras=None, domain=None):
         return post_case_blocks(case_blocks, form_extras=form_extras, domain=domain)
 
+    def get_case(self, case_id):
+        return self.case_model.get(case_id)
+
     def store_attachments(self, xform, attachments):
         """
         Takes a list of Attachment namedtuples with content, name, and content_type and stores them to the XForm

--- a/corehq/form_processor/interfaces/processor.py
+++ b/corehq/form_processor/interfaces/processor.py
@@ -85,6 +85,12 @@ class FormProcessorInterface(object):
     def get_case(self, case_id):
         return self.case_model.get(case_id)
 
+    def get_cases(self, case_ids):
+        return self.case_model.get_cases(case_ids)
+
+    def get_case_xform_ids(self, case_id):
+        return self.case_model.get_case_xform_ids(case_id)
+
     def store_attachments(self, xform, attachments):
         """
         Takes a list of Attachment namedtuples with content, name, and content_type and stores them to the XForm

--- a/corehq/form_processor/tests/test_basic_cases.py
+++ b/corehq/form_processor/tests/test_basic_cases.py
@@ -30,7 +30,7 @@ class FundamentalCaseTests(TestCase):
             }
         )
 
-        case = self.interface.case_model.get(case_id)
+        case = self.interface.get_case(case_id)
         self.assertIsNotNone(case)
         self.assertEqual(case.case_id, case_id)
         self.assertEqual(case.owner_id, 'owner1')
@@ -71,7 +71,7 @@ class FundamentalCaseTests(TestCase):
             }
         )
 
-        case = self.interface.case_model.get(case_id)
+        case = self.interface.get_case(case_id)
         self.assertEqual(case.owner_id, 'owner2')
         self.assertEqual(case.name, 'update_case')
         self.assertEqual(case.opened_on, opened_on)
@@ -98,7 +98,7 @@ class FundamentalCaseTests(TestCase):
             False, case_id, user_id='user2', date_modified=modified_on, close=True
         )
 
-        case = self.interface.case_model.get(case_id)
+        case = self.interface.get_case(case_id)
         self.assertEqual(case.owner_id, 'owner1')
         self.assertEqual(case.modified_on, modified_on)
         self.assertEqual(case.modified_by, 'user2')
@@ -124,7 +124,7 @@ class FundamentalCaseTests(TestCase):
             }
         )
 
-        case = self.interface.case_model.get(child_case_id)
+        case = self.interface.get_case(child_case_id)
         self.assertEqual(len(case.indices), 1)
         index = case.indices[0]
         self.assertEqual(index.identifier, 'mom')
@@ -148,7 +148,7 @@ class FundamentalCaseTests(TestCase):
             }
         )
 
-        case = self.interface.case_model.get(child_case_id)
+        case = self.interface.get_case(child_case_id)
         self.assertEqual(case.indices[0].identifier, 'mom')
 
         _submit_case_block(
@@ -156,7 +156,7 @@ class FundamentalCaseTests(TestCase):
                 'mom': ('other_mother', mother_case_id)
             }
         )
-        case = self.interface.case_model.get(child_case_id)
+        case = self.interface.get_case(child_case_id)
         self.assertEqual(case.indices[0].referenced_type, 'other_mother')
 
     @run_with_all_backends
@@ -175,7 +175,7 @@ class FundamentalCaseTests(TestCase):
             }
         )
 
-        case = self.interface.case_model.get(child_case_id)
+        case = self.interface.get_case(child_case_id)
         self.assertEqual(len(case.indices), 1)
 
         _submit_case_block(
@@ -183,7 +183,7 @@ class FundamentalCaseTests(TestCase):
                 'mom': ('mother', '')
             }
         )
-        case = self.interface.case_model.get(child_case_id)
+        case = self.interface.get_case(child_case_id)
         self.assertEqual(len(case.indices), 0)
 
     def test_case_with_attachment(self):

--- a/corehq/form_processor/tests/test_dbcache.py
+++ b/corehq/form_processor/tests/test_dbcache.py
@@ -74,7 +74,7 @@ class CaseDbCacheTest(TestCase):
             self.assertFalse(cache.in_cache(id))
 
         for id in case_ids:
-            cache.set(id, self.interface.case_model.get(id))
+            cache.set(id, self.interface.get_case(id))
 
         for i, id in enumerate(case_ids):
             self.assertTrue(cache.in_cache(id))

